### PR TITLE
add native filters for "(filter) is true" and "(filter) is false"

### PR DESCRIPTION
**Pull request by clintropolis**
### Description

changes:
* add IsTrueDimFilter, IsFalseDimFilter, and abstract IsBooleanDimFilter for native json filter implementations of `(filter) IS TRUE` and `(filter) IS FALSE`
* add IsBooleanFilter for actual filtering logic for these filters, which ignore includeUnknown to always use matches with false for true and !matches with true for false
* fix test incorrectly adjusted to wrong answer in #15058

This PR has:

- [ ] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java [hunk 1]
```diff
@@ -488,13 +512,23 @@ private static DimFilter toSimpleLeafFilter(
...
+      // legacy behavior
       return toSimpleLeafFilter(
           plannerContext,
           rowSignature,
           virtualColumnRegistry,
           Iterables.getOnlyElement(((RexCall) rexNode).getOperands())
       );
     } else if (kind == SqlKind.IS_FALSE || kind == SqlKind.IS_NOT_TRUE) {
+      if (NullHandling.useThreeValueLogic()) {
+        // use expression filter to get isfalse or nottrue expressions for correct 3vl behavior
```
Comment (soumyava) [inline sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java:528]: nit. isnottrue in place of nottrue

**Review by soumyava — COMMENTED** [review]
Took 1 pass, looks good to me so far. Will take another pass tomorrow

File: sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java (hunk 1 above)
Comment (clintropolis) [reply sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java:528]: ah, i was being consistent with the native expression names:
* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/math/expr/Function.java#L2402
* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/math/expr/Function.java#L2434

* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/math/expr/Function.java#L2338
* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/math/expr/Function.java#L2370

**Review by soumyava — APPROVED** [review]
Took a 2nd pass. Checked it out and ran some test cases around it with nullif. LGTM !

**Review by pranavbhole — APPROVED** [review]

**Pull request merged**